README.md

```diff
@@ -1,5 +1,40 @@
 # SourceCred Research
 
-This repository is used to keep track of SourceCred research matters.
+This repository contains research infrastructure, code, and results for SourceCred.
 
-We will soon add a description of the attribution algorithm, and open research problems.
+## Setup
+
+Our research codebase is written in Python3, and makes use of Jupyter notebooks.
+We recommend using [Conda] to create and manage Python environments. See the
+[Conda installation guide].
+
+[Conda]: https://conda.io/en/latest/
+[Conda installation guide]: https://docs.conda.io/projects/conda/en/latest/user-guide/install/
+
+This repository uses [Black] as a code formatter, and [pre-commit] to ensure that all code is
```

Comment:
Pre-commit hooks don’t actually ensure anything. For instance, they
don’t run on `git rebase --continue` even if you’ve made changes to the
commit and it will in fact be re-committed. And some people just don’t
like them (I’m not a huge fan myself) and won’t set them up.

All this to say—if you want reliable formatting, you need tests on CI to
that effect. :-)

Reply:
Got it. Maybe for these reasons I should just ditch the pre-commit plan and go with CI tests (which we would need anyway). 

(Or we could include pre-commit "optionally" for those who want to install it, but CI acts as the failsafe that covers everyone. Personally, I now have black integrated into neovim, so I don't really care either way.)